File: lib/person.js
'use strict';

var Person = function(h) {
  Object.keys(h).forEach(k => this[k] = h[k]);
};

Person.prototype.age = function() {
  var dob = new Date(this.dob);
  var today = new Date();
  var thisYear = today.getFullYear();
  if (dob.getMonth() > today.getMonth() ||
      dob.getMonth() === today.getMonth() &&
      dob.getDate() >= today.getDate()) {
    thisYear -= 1;
  }
  return thisYear - dob.getFullYear();
};

module.exports = Person;

File: lib/person.rb
#
require 'date'
#
class Person
  def initialize(h)
    h.keys.each do |key|
      instance_variable_set(('@' + key.to_s).to_sym, h[key])
      self.class.send(:attr_reader, key)
    end
  end

  def age
    dob = Date.parse(@dob)
    today = Date.today
    if dob.month > today.month ||
       dob.month == today.month && dob.day >= today.day
      today <<= 12 # one year younger
    end
    today.year - dob.year
  end
end
